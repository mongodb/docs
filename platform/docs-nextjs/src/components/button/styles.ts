import { css } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';

const focusBoxShadow = (color: string) => `
  0 0 0 2px ${color}, 
  0 0 0 4px ${palette.blue.light1};
`;

// Theme-specific styles were copied from the original Button component
// to allow pre-rendering for dark mode
export const disabledStyle = css`
  &[aria-disabled='true'] {
    &,
    &:hover,
    &[data-hover='true'],
    &:active,
    &[data-active='true'] {
      background-color: ${palette.gray.light2};
      border-color: ${palette.gray.light1};
      color: ${palette.gray.base};
      box-shadow: none;
      cursor: not-allowed;
    }

    &:focus-visible,
    &[data-focus='true'] {
      color: ${palette.gray.base};
      box-shadow: ${focusBoxShadow(palette.white)};
    }

    // Needs to be nested here to ensure proper cascading
    .dark-theme & {
      &,
      &:hover,
      &[data-hover='true'],
      &:active,
      &[data-active='true'] {
        background-color: ${palette.gray.dark3};
        border-color: ${palette.gray.dark2};
        color: ${palette.gray.dark1};
        box-shadow: none;
        cursor: not-allowed;
      }

      &:focus-visible,
      &[data-focus='true'] {
        color: ${palette.gray.dark1};
        box-shadow: ${focusBoxShadow(palette.black)};
      }
    }
  }
`;
