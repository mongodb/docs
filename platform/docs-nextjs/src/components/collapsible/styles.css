/* minor css to change nth child properties in browsers without support for :nth-child(n of [selector]) */

.collapsible {
  border-top: 1px solid #e8edeb;
  margin-top: 32px;
}

.collapsible ~ .collapsible {
   border-top: none;
   margin-top: 0;
}