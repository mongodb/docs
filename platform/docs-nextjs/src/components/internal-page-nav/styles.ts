import { css } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';

export const navLinkButtonStyle = css`
  background-color: ${palette.gray.light3};
  color: ${palette.black};

  .dark-theme & {
    background-color: ${palette.gray.dark2};
    color: ${palette.white};
  }
`;
