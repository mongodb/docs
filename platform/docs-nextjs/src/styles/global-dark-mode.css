body {
  --font-color-primary: var(--black);
  --font-color-light: var(--gray-base);
  --heading-color-primary: var(--green-dark2);
  --opposite-color: var(--black);
  --font-color-active: var(--green-dark2);

  --link-color-primary: var(--blue-base);
  --link-font-weight: 500;

  --background-color-primary: var(--white);
  --background-color-secondary: var(--gray-light3);

  --sidenav-bg-color: var(--gray-light3);
  --sidenav-border-bottom-color: var(--gray-light2);

  --sidenav-toggle-color: var(--green-dark2);
  --sidenav-toggle-border-color: var(--gray-light2);

  --sidenav-item-title: var(--gray-dark3);
  --sidenav-item-color: var(--black);
  --sidenav-hover-bg-color: var(--gray-light2);
  --sidenav-active-bg-color: var(--green-light3);
  --sidenav-active-color: var(--green-dark2);
  --sidenav-active-before-color: var(--green-dark1);

  --select-button-bg-color: var(--white);
  --select-button-color: var(--black);
  --select-button-carot: var(--gray-dark2);

  /* --tab-color-primary also accounts for the Docs Home button */
  --tab-color-primary: var(--gray-dark1);
  --tab-color-secondary: var(--green-dark2);

  --wayfinding-bg-color: var(--gray-light3);
  --wayfinding-border-color: var(--gray-light2);
}

.dark-theme body {
  --font-color-primary: var(--gray-light2);
  --font-color-light: var(--gray-light2);
  --heading-color-primary: var(--gray-light2);
  --opposite-color: var(--white);
  --font-color-active: var(--green-light1);

  --link-color-primary: var(--blue-light1);
  --link-font-weight: 700;

  --background-color-primary: var(--black);
  --background-color-secondary: transparent;

  --sidenav-bg-color: var(--gray-dark4);
  --sidenav-border-bottom-color: var(--gray-dark2);

  --sidenav-toggle-color: var(--gray-light1);
  --sidenav-toggle-border-color: var(--gray-dark2);

  --sidenav-item-title: var(--gray-light2);
  --sidenav-item-color: var(--gray-light2);
  --sidenav-hover-bg-color: var(--gray-dark3);
  --sidenav-active-bg-color: var(--green-dark3);
  --sidenav-active-color: var(--white);
  --sidenav-active-before-color: var(--green-base);

  --select-button-bg-color: var(--gray-dark4);
  --select-button-color: var(--gray-light3);
  --select-button-carot: var(--gray-light1);

  --tab-color-primary: var(--gray-light1);
  --tab-color-secondary: var(--gray-light2);

  --wayfinding-bg-color: var(--gray-dark3);
  --wayfinding-border-color: var(--gray-dark2);

  /* Style the scrollbar background */
  ::-webkit-scrollbar {
    background: var(--gray-dark3);
  }

  /* Style the scrollbar handle */
  ::-webkit-scrollbar-thumb {
    background: var(--gray-dark2);
    border: 3px solid transparent;
    border-radius: 9px;
    background-clip: content-box;
  }
}

body {
  background-color: var(--background-color-primary);
  color: var(--font-color-primary);
}

.nav-dark {
  display: none;
}

.nav-light {
  display: initial;
}

.dark-theme .nav-dark {
  display: initial;
}

.dark-theme .nav-light {
  display: none;
}
