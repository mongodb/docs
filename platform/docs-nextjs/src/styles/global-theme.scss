// theme.module.scss

$font-size-xsmall: 11px;
$font-size-tiny: 12px;
$font-size-small: 13px;
$font-size-default: 16px;
$font-size-h1: 36px;
$font-size-h2: 24px;
$font-size-h3: 18px;
$font-size-h4: 16px;

$size-tiny: 4px;
$size-small: 8px;
$size-default: 16px;
$size-medium: 24px;
$size-large: 32px;
$size-xlarge: 64px;
$size-xxlarge: 128px;
$size-max-width: 1400px;

// Breakpoints
$breakpoints-xSmall: 320;
$breakpoints-small: 480;
$breakpoints-medium: 767;
$breakpoints-large: 1024;
$breakpoints-xLarge: 1200;
$breakpoints-xxLarge: 1440;
$breakpoints-xxxLarge: 1920;

// Screen Sizes
$screenSize-upToXSmall: "only screen and (max-width: #{$breakpoints-xSmall}px)";
$screenSize-xSmallAndUp: "not all and (max-width: #{$breakpoints-xSmall}px)";
$screenSize-upToSmall: "only screen and (max-width: #{$breakpoints-small}px)";
$screenSize-smallAndUp: "not all and (max-width: #{$breakpoints-small}px)";
$screenSize-upToMedium: "only screen and (max-width: #{$breakpoints-medium}px)";
$screenSize-mediumAndUp: "not all and (max-width: #{$breakpoints-medium}px)";
$screenSize-upToLarge: "only screen and (max-width: #{$breakpoints-large}px)";
$screenSize-largeAndUp: "not all and (max-width: #{$breakpoints-large}px)";
$screenSize-upToXLarge: "only screen and (max-width: #{$breakpoints-xLarge}px)";
$screenSize-xLargeAndUp: "not all and (max-width: #{$breakpoints-xLarge}px)";
$screenSize-upTo2XLarge: "only screen and (max-width: #{$breakpoints-xxLarge}px)";
$screenSize-2XLargeAndUp: "not all and (max-width: #{$breakpoints-xxLarge}px)";
$screenSize-upTo3XLarge: "only screen and (max-width: #{$breakpoints-xxxLarge}px)";
$screenSize-3XLargeAndUp: "not all and (max-width: #{$breakpoints-xxxLarge}px)";
$screenSize-tablet: "only screen and (min-width: #{($breakpoints-small + 1)}px) and (max-width: #{$breakpoints-large}px)";

// Header heights on different screens
$header-banner-height: 40px;
$header-navbar-height: 95px;
$header-navbar-mobile-height: 56px;
$header-docs-mobile-menu-height: 52px;
$header-navbar-scroll-offset: 85px;
$header-action-bar-mobile-height: 60px;
$header-composable-desktop-height: 90px;
$header-composable-mobile-height: 190px;

$widgets-button-container-mobile-height: 60px;

$transition-speed-ia-exit: 100ms;
$transition-speed-ia-enter: 200ms;
$transition-speed-content-fade: 300ms;

$zindex-content: 1;
$zindex-action-bar: 800;
$zindex-sidenav: 900;
$zindex-header: 1000;
$zindex-widgets: 2000;
$zindex-popovers: 10000;

$color-pref-dark: '(prefers-color-scheme: dark)';
$color-pref-light: '(prefers-color-scheme: light)';
$color-pref-auto: '(prefers-color-scheme: auto)';

$screen-tablet: 'only screen and (min-width: 481px) and (max-width: 1024px)';