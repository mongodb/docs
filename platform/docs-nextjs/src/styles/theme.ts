const fontSize = {
  xsmall: '11px',
  tiny: '12px',
  small: '13px',
  default: '16px',
  h1: '36px',
  h2: '24px',
  h3: '18px',
  h4: '16px',
};

const size = {
  tiny: '4px',
  small: '8px',
  default: '16px',
  medium: '24px',
  large: '32px',
  xlarge: '64px',
  xxlarge: '128px',
  maxWidth: '1400px',
  stripUnit(unit: string): number {
    return parseInt(unit, 10);
  },
};

/**
 * store common responsive sizes as numbers
 */
const breakpoints = {
  xSmall: 320,
  small: 480,
  medium: 767,
  large: 1024,
  xLarge: 1200,
  xxLarge: 1440,
  xxxLarge: 1920,
};

/**
 * store common responsive sizes
 */
const screenSize = {
  upToXSmall: `only screen and (max-width: ${breakpoints.xSmall}px)`,
  xSmallAndUp: `not all and (max-width: ${breakpoints.xSmall}px)`,
  upToSmall: `only screen and (max-width: ${breakpoints.small}px)`,
  smallAndUp: `not all and (max-width: ${breakpoints.small}px)`,
  upToMedium: `only screen and (max-width: ${breakpoints.medium}px)`,
  mediumAndUp: `not all and (max-width: ${breakpoints.medium}px)`,
  upToLarge: `only screen and (max-width: ${breakpoints.large}px)`,
  largeAndUp: `not all and (max-width: ${breakpoints.large}px)`,
  upToXLarge: `only screen and (max-width: ${breakpoints.xLarge}px)`,
  xLargeAndUp: `not all and (max-width: ${breakpoints.xLarge}px)`,
  upTo2XLarge: `only screen and (max-width: ${breakpoints.xxLarge}px)`,
  '2XLargeAndUp': `not all and (max-width: ${breakpoints.xxLarge}px)`,
  upTo3XLarge: `only screen and (max-width: ${breakpoints.xxxLarge}px)`,
  '3XLargeAndUp': `not all and (max-width: ${breakpoints.xxxLarge}px)`,
  tablet: `only screen and (min-width: ${breakpoints.small + 1}px) and (max-width: ${breakpoints.large}px)`,
};

const header = {
  bannerHeight: '40px',
  navbarHeight: '95px',
  navbarMobileHeight: '56px',
  docsMobileMenuHeight: '52px',
  // used for scrolling elements into place, considering sticky header
  navbarScrollOffset: '85px',
  actionBarMobileHeight: '60px',
  composableDesktopHeight: '90px',
  composableMobileHeight: '190px',
};

const widgets = {
  buttonContainerMobileHeight: '60px',
};

const transitionSpeed = {
  iaExit: '100ms',
  iaEnter: '200ms',
  contentFade: '300ms',
};

// z-indexes for topmost major elements on the site. Subcomponents should have their own z-indexes set
const zIndexes = {
  content: 1,
  actionBar: 800,
  sidenav: 900,
  header: 1000,
  widgets: 2000,
  popovers: 10000,
};

// media queries that define system color preference
const colorPreference = {
  dark: '(prefers-color-scheme: dark)',
  light: '(prefers-color-scheme: light)',
  auto: '(prefers-color-scheme: auto)',
};

export const theme = {
  breakpoints,
  fontSize,
  header,
  widgets,
  screenSize,
  size,
  transitionSpeed,
  zIndexes,
  colorPreference,
};
