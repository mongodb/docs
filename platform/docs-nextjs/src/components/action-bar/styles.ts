import { css } from '@leafygreen-ui/emotion';
import styled from '@emotion/styled';
import { theme } from '@/styles/theme';
import { palette } from '@leafygreen-ui/palette';
import { displayNone } from '@/utils/display-none';

export const ActionBarSearchContainer = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  background: inherit;

  @media ${theme.screenSize.mediumAndUp} {
    padding-right: ${theme.size.default};
  }

  @media ${theme.screenSize.upToLarge} {
    max-width: unset;
    justify-content: space-between;
    width: 100%;
  }
`;

interface StyledInputProps {
  sidenav?: boolean;
  mobileSearchActive: boolean;
}

export const StyledInputContainer = styled.div<StyledInputProps>`
  width: 100%;
  max-width: 610px;
  background: inherit;

  div[role='searchbox'] {
    background-color: var(--search-input-background-color);
    width: 100%;
  }

  --search-input-background-color: ${palette.white};
  .dark-theme & {
    --search-input-background-color: ${palette.gray.dark4};
  }

  @media ${theme.screenSize.mediumAndUp} {
    width: ${({ sidenav }) => (sidenav ? '70' : '100')}%;
  }

  ${(props) => {
    return (
      props.mobileSearchActive &&
      `
      display: flex !important;
      position: absolute;
      width: calc(100% - ${theme.size.medium} - ${theme.size.medium});
      z-index: 40;
      max-width: unset;
      left: ${theme.size.medium};
      column-gap: ${theme.size.medium};
      align-items: center;

      form {
        width: 100%;
        margin-right: ${theme.size.medium};
      }
    `
    );
  }}
`;

// Used to ensure dropdown is same width as input
export const StyledSearchBoxRef = styled.div`
  width: 100%;
`;

export const searchInputStyling = ({ mobileSearchActive }: StyledInputProps) => css`
  ${!mobileSearchActive && displayNone.onMedium};

  @media ${theme.screenSize.upToMedium} {
    input[type='search'] {
      font-size: ${theme.fontSize.default};
    }
  }

  ${mobileSearchActive &&
  `
    display: flex !important;
    width: calc(100% - ${theme.size.medium} - ${theme.size.medium});
    z-index: 40;
    max-width: unset;
    background: var(--background-color-primary);
    align-items: center;

    > form {
      width: 100%;
      margin-right: ${theme.size.medium};
    }
  `}
`;

export const ActionsBox = styled('div')`
  display: flex;
  align-items: center;
  column-gap: ${theme.size.default};
  position: relative;
  top: 0;
  margin: 0 ${theme.size.large} 0 ${theme.size.medium};
  justify-self: flex-end;
  grid-column: -2/-1;

  @media ${theme.screenSize.upToLarge} {
    column-gap: 6px;
    margin-right: ${theme.size.medium};
    margin-left: ${theme.size.small};
  }

  @media ${theme.screenSize.upToMedium} {
    margin-left: 1px;
  }
`;

export const overlineStyling = css`
  display: flex;
  flex-direction: row;
  align-items: center;
  color: ${palette.gray.dark1};
  text-transform: uppercase;
  text-wrap: nowrap;
  font-weight: 600;
  cursor: pointer;
  ${displayNone.onLargerThanTablet};
  font-size: ${theme.fontSize.tiny};
  padding-right: ${theme.size.large};
  .dark-theme & {
    color: ${palette.gray.light2};
  }

  > svg {
    margin-right: ${theme.size.small};
  }
`;

export const searchIconStyling = css`
  ${displayNone.onLargerThanMedium};
  float: right;
  justify-content: right;
`;

export const offlineStyling = css`
  @media ${theme.screenSize.largeAndUp} {
    display: none;
  }
`;

const hideOnEnLang = `
  &:not(:lang(EN)) {
    display: none;
  }
`;

export const chatbotButtonStyling = css`
  text-wrap-mode: nowrap;
  ${displayNone.onMobileAndTablet};
  ${hideOnEnLang}
`;

export const chatbotMobileButtonStyling = css`
  ${displayNone.onLargerThanTablet}
  ${hideOnEnLang}
  color: ${palette.green.dark2};

  .dark-theme & {
    color: ${palette.green.dark1};
  }
`;
