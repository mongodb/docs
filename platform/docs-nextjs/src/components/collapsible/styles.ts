import { css } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';
import { theme } from '@/styles/theme';

export const collapsibleStyle = css`
  border-bottom: 1px solid ${palette.gray.light2};

  /* minor css to change nth child properties in browsers without support for :nth-child(n of [selector]) */

  & {
    border-top: 1px solid #e8edeb;
    margin-top: 32px;
  }

  & ~ & {
    border-top: none;
    margin-top: 0;
  }

  &:nth-child(1 of .collapsible) {
    border-top: 1px solid ${palette.gray.light2};
    margin-top: ${theme.size.large};
  }

  &:nth-last-child(1 of .collapsible) {
    margin-bottom: ${theme.size.large};
  }
`;

export const headerContainerStyle = css`
  display: flex;
  justify-content: space-between;
  column-gap: ${theme.size.medium};
  margin: ${theme.size.large} 0;
`;

export const headerStyle = css`
  margin-top: 0;
`;

export const iconStyle = css`
  align-self: center;
  flex-shrink: 0;
`;

export const innerContentStyle = css`
  overflow: hidden;
  height: 0;
  color: --font-color-primary;

  [aria-expanded='true'] & {
    height: auto;
  }
`;
