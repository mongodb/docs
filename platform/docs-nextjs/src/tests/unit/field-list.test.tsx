import FieldList from '@/components/field-list';
import { ThemeProvider } from '@emotion/react';
import { theme } from '@/styles/theme';
import { render } from '@testing-library/react';
import type { FieldListNode } from '@/types/ast';

import mockData from '../data/field-list.test.json';
const typedMockData = mockData as FieldListNode;

it('renders correctly', () => {
  const tree = render(
    <ThemeProvider theme={theme}>
      <FieldList nodeChildren={typedMockData.children} />
    </ThemeProvider>,
  );
  expect(tree.asFragment()).toMatchSnapshot();
});
