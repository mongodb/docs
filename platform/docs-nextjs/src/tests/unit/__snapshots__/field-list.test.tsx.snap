// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-spacing: 0;
  font-size: 13px;
  line-height: 24px;
  margin: 24px 0;
  max-width: 100%;
}

.emotion-0 tbody {
  vertical-align: top;
}

.emotion-1 {
  font-size: 16px;
}

.emotion-1>th,
.emotion-1>td {
  padding: 11px 5px 12px;
  text-align: left;
}

.emotion-2 {
  font-size: 13px;
  line-height: 20px;
  display: inline;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  border-radius: 3px;
  font-family: 'Source Code Pro',Menlo,monospace;
  line-height: 20px;
  background-color: #F9FBFA;
  border: 1px solid #E8EDEB;
  color: #1C2D38;
  white-space: nowrap;
  font-size: unset;
  display: inline;
  color: var(--font-color-primary);
  background: var(--background-color-secondary);
  word-wrap: break-word;
  white-space: unset;
}

.lg-ui-0000:hover>.emotion-2 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.lg-ui-0000:hover>.emotion-2 {
  box-shadow: 0 0 0 3px #E8EDEB;
  border: 1px solid #C1C7C6;
}

.lg-ui-0000:focus-visible>.emotion-2 {
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC;
  border: 1px solid #016BF8;
}

a .emotion-2 {
  color: inherit;
}

<table
    class="emotion-0"
  >
    <colgroup>
      <col
        class="field-name"
      />
      <col
        class="field-body"
      />
    </colgroup>
    <tbody>
      <tr
        class="emotion-1"
      >
        <th>
          Returns:
        </th>
        <td>
          A JSON Web Token string encoded for the provided 
          <code
            class="emotion-2"
          >
            payload
          </code>
          .
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
